Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt

```diff
@@ -0,0 +1,54 @@
+package com.tabnineCommon.chat.commandHandlers.context
+
+import com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.util.Processor
+import java.awt.Point
+
+data class DiagnosticsContext(
+    private val diagnosticsText: String? = null,
+) : EnrichingContextData {
+    // Used for serialization - do not remove
+    private val type: EnrichingContextType = EnrichingContextType.Diagnostics
+
+    companion object {
+        fun create(editor: Editor, project: Project): DiagnosticsContext? {
+            val visibleRange = getVisibleRange(editor) ?: return null
+            val highlights = mutableListOf<String>()
+
+            DaemonCodeAnalyzerImpl.processHighlights(
```

Comment:
can you check if this can throw an exception ?

Reply:
it doesn't say so in the method declaration, but probably can throw an exception if you don't have read access or something.
anyway, if it throws an exception, we will send an error message to the chat and the chat knows how to handle it.
good enough IMO, wdyt?